### PR TITLE
Move to intra-doc links for /library/core/src/intrinsics.rs

### PR DESCRIPTION
**Pull request by denisvasilik**
Helps with #75080.

@rustbot modify labels: T-doc, A-intra-doc-links, T-rustdoc

Known issues:

* The following f32 and f64 primitive methods cannot be resolved:

    f32/f64::powi
    f32/f64::sqrt
    f32/f64::sin
    f32/f64::cos
    f32/f64::powf
    f32/f64::exp
    f32/f64::exp2
    f32/f64::ln
    f32/f64::log2
    f32/f64::log10
    f32/f64::mul_add
    f32/f64::abs
    f32/f64::copysign
    f32/f64::floor
    f32/f64::ceil
    f32/f64::trunc
    f32/f64::round

* Links from core to std:

    [`std::pointer::*`]
    [`std::process::abort`]
    [`from_raw_parts`]
    [`Vec::append`]

* Links with anchors?

	I provided a separate commit that replaced links with anchors by intra-doc links.
	Here the anchor location information gets lost, so its questionable whether to 
	actually replace those links.

### CODE REVIEW

**Comment by jyn514** [comment]
> Here the anchor location information gets lost, so its questionable whether to
actually replace those links.

You can add the anchors to the end of the intra-doc links and they'll still resolve.

**Comment by jyn514** [comment]
> The following f32 and f64 primitive methods cannot be resolved

That seems weird ... `f32` itself resolves fine. Might be related to https://github.com/rust-lang/rust/pull/75649.

**Comment by jyn514** [comment]
Don't have time to review this fully so will defer to @LukasKalbertodt, but from the description you wrote this seems fine after you fix the anchor links.

**Comment by jyn514** [comment]
r? @jyn514 

I found the time 😆

File: library/core/src/intrinsics.rs [hunk 1]
```diff
@@ -72,7 +72,7 @@ extern "rust-intrinsic" {
     ///
     /// The stabilized version of this intrinsic is available on the
     /// `std::sync::atomic` types via the `compare_exchange` method by passing
-    /// [`Ordering::SeqCst`](../../std/sync/atomic/enum.Ordering.html)
+    /// [`Ordering::SeqCst`](crate::sync::atomic::Ordering::SeqCst)
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:75]: This is a _lot_ of `crate::sync::atomic::Ordering` throughout the file. Maybe it makes sense to `#[allow(unused_imports)] use crate::sync::atomic::Ordering;` at the top with a comment that it's for intra-doc links? Then you could remove these links altogether.

File: library/core/src/intrinsics.rs [hunk 2]
```diff
@@ -280,23 +280,23 @@ extern "rust-intrinsic" {
     /// `std::sync::atomic` types via the `load` method by passing
     /// [`Ordering::SeqCst`](crate::sync::atomic::Ordering::SeqCst)
     /// as the `order`. For example,
-    /// [`AtomicBool::load`](../../std/sync/atomic/struct.AtomicBool.html#method.load).
+    /// [`AtomicBool::load`](crate::sync::atomic::AtomicBool::load).
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:283]: Same comment about `use crate::sync::atomic::AtomicBool;`.

File: library/core/src/intrinsics.rs [hunk 3]
```diff
@@ -373,39 +373,39 @@ extern "rust-intrinsic" {
     /// `std::sync::atomic` types via the `fetch_add` method by passing
     /// [`Ordering::SeqCst`](crate::sync::atomic::Ordering::SeqCst)
     /// as the `order`. For example,
-    /// [`AtomicIsize::fetch_add`](../../std/sync/atomic/struct.AtomicIsize.html#method.fetch_add).
+    /// [`AtomicIsize::fetch_add`](crate::sync::atomic::AtomicIsize::fetch_add).
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:376]: Same comment about `use crate::sync::atomic::AtomicIsize;`.

File: library/core/src/intrinsics.rs [hunk 4]
```diff
@@ -619,39 +619,39 @@ extern "rust-intrinsic" {
     /// `std::sync::atomic` signed integer types via the `fetch_max` method by passing
     /// [`Ordering::SeqCst`](crate::sync::atomic::Ordering::SeqCst)
     /// as the `order`. For example,
-    /// [`AtomicI32::fetch_max`](../../std/sync/atomic/struct.AtomicI32.html#method.fetch_max).
+    /// [`AtomicI32::fetch_max`](crate::sync::atomic::AtomicI32::fetch_max).
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:622]: Same comment about `use crate::sync::atomic::AtomicI32;`

File: library/core/src/intrinsics.rs [hunk 5]
```diff
@@ -701,39 +701,39 @@ extern "rust-intrinsic" {
     /// `std::sync::atomic` unsigned integer types via the `fetch_min` method by passing
     /// [`Ordering::SeqCst`](crate::sync::atomic::Ordering::SeqCst)
     /// as the `order`. For example,
-    /// [`AtomicU32::fetch_min`](../../std/sync/atomic/struct.AtomicU32.html#method.fetch_min).
+    /// [`AtomicU32::fetch_min`](crate::sync::atomic::AtomicU32::fetch_min).
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:704]: Same comment about `use crate::sync::atomic::AtomicU32;`.

File: library/core/src/intrinsics.rs [hunk 6]
```diff
@@ -823,31 +823,31 @@ extern "rust-intrinsic" {
     /// An atomic fence.
     ///
     /// The stabilized version of this intrinsic is available in
-    /// [`std::sync::atomic::fence`](../../std/sync/atomic/fn.fence.html)
+    /// [`std::sync::atomic::fence`](crate::sync::atomic::fence)
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:826]: Same comment about `use crate::sync::atomic`.

File: library/core/src/intrinsics.rs [hunk 7]
```diff
@@ -2181,8 +2179,8 @@ pub unsafe fn copy_nonoverlapping<T>(src: *const T, dst: *mut T, count: usize) {
 /// Note that even if the effectively copied size (`count * size_of::<T>()`) is
 /// `0`, the pointers must be non-NULL and properly aligned.
 ///
-/// [`Copy`]: ../marker/trait.Copy.html
-/// [`read`]: ../ptr/fn.read.html
+/// [`Copy`]: crate::marker::Copy
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:2182]: I think `Copy` is in the prelude.
```suggestion
```

File: library/core/src/intrinsics.rs [hunk 8]
```diff
@@ -2078,8 +2077,8 @@ pub(crate) fn is_nonoverlapping<T>(src: *const T, dst: *const T, count: usize) -
...
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:2080]:
```suggestion
```

File: library/core/src/intrinsics.rs [hunk 9]
```diff
@@ -76,7 +76,7 @@ extern "rust-intrinsic" {
     /// as both the `success` and `failure` parameters. For example,
     /// [`AtomicBool::compare_exchange`][compare_exchange].
     ///
-    /// [compare_exchange]: ../../std/sync/atomic/struct.AtomicBool.html#method.compare_exchange
+    /// [compare_exchange]: crate::sync::atomic::AtomicBool::compare_exchange
```
Comment (jyn514) [inline library/core/src/intrinsics.rs:79]: Same comment about `use crate::sync::atomic::AtomicBool`.

**Review by jyn514 — COMMENTED** [review]
It would also be nice to change ``` `crate::sync::atomic` ``` to a link: `[crate::sync::atomic]`.

File: library/core/src/intrinsics.rs (hunk 1 above)
Comment (denisvasilik) [reply library/core/src/intrinsics.rs:75]: Sounds good to me.

**Comment by denisvasilik** [comment]
Pretty good suggestions, this cleans up the file a bit. I will integrate the changes and come up with another commit. Thanks.

**Pull request merged**